### ISSUE TITLE
Documentation appears to be out of date. 

### ISSUE THREAD

**Issue by ForeverWintr** [GeospatialPython/pyshp#6]
The documentation for `shapefile.Writer` says that it has an `autobalance` attribute and `Balance()` method, neither of which seem to exist (they do exist in Editor though, I see).

**Comment by geographika** [comment]
Also the example for iterShapeRecords:

>>> shapeRecs = sf.iterShapeRecords()
>>> for shape, rec in shapeRecs:
...     # do something here

produces:

``TypeError: iteration over non-sequence``

It should be:

    for sr in sf.iterShapeRecords():

        s = sr.shape
        r = sr.record

**Comment by billgale** [comment]
I would like to reiterate that this documentation for iterShapeRecords() 

>>> shapeRecs = sf.iterShapeRecords()
>>> for shape, rec in shapeRecs:
...     # do something here

still causes an exception as of release pyshp 1.2.3:

        for shape, record in sf.iterShapeRecords():
    TypeError: iteration over non-sequence

Here is the one-liner I used to get around it, but I prefer geographika's solution as it requires no external package.

```python
import itertools

for shape, record in itertools.izip(sf.iterShapes(), sf.iterRecords()):
# for sr in sf.iterShapeRecords():
     # shape = sr.shape
     # record = sr.record
```

**Comment by karimbahgat** [comment]
Fixed in 6906953acd2caf2d4d6890160344e9d2baea058b.
